src/standalone/api/kernels/backgroundExecution.ts

```diff
@@ -10,6 +10,19 @@ import { raceCancellation } from '../../../platform/common/cancellation';
 import { getNotebookCellOutputMetadata } from '../../../kernels/execution/helpers';
 import { unTrackDisplayDataForExtension } from '../../../kernels/execution/extensionDisplayDataTracker';
 import { traceWarning } from '../../../platform/logging';
+import { IBackgroundThreadService } from '../../../kernels/jupyter/types';
+import { injectable } from 'inversify';
+
+@injectable()
+export class BackgroundThreadService implements IBackgroundThreadService {
+    execCodeInBackgroundThread<T>(
+        kernel: IKernel,
+        codeWithReturnStatement: string[],
+        token: CancellationToken
+    ): Promise<T | undefined> {
+        return execCodeInBackgroundThread(kernel, codeWithReturnStatement, token);
```

Comment:
Just noticed this PR, and wanted to provide some quick feedback,
We try to avoid injections where possible, as this slows the extension startup.
In this case you can just keep the class, but remove teh `@injectable decorator
& construct the class when you need it, or just pass it from the calling code
Or remove the class entirely and just use the function, I'm assuming you need the class for mocking/testing purposes
Else having this class is not necessary, as it just wraps a function

For instance today we don't use this class in other places, but use the function directly.
Personally prefer to remove this class and keep the function, else there are two ways of doing the same thing 

@rebornix /cc thoughts?

Reply:
I need it to work around import zone restrictions - the function is in `standalone` (which doesn't seem right to me, but it depends on the API code which is also in that zone), and the caller is in kernel. Neither of those zones can depend on the other.

Reply:
+1 for avoiding `injectable` if possible. This is a static helper function that can be optimized by bundlers, making it an `injectable` function doesn't seem to give us any benefit at this moment.

Reading the dependencies of `execCodeInBackgroundThread`, it seems we can move it into `kernel` component. `src\kernels\execution\helpers.ts` might be a good place.

Reply:
I missed that `const api = createKernelApiForExtension(JVSC_EXTENSION_ID, kernel);` which depends on `NotebookKernelExecution#onDidReceiveDisplayUpdate`. It needs some code movement to make it happen. Like @DonJayamanne suggested, probably we leave it to him for the refactoring.